### DISCUSSION TITLE
elasticsearch 7 NoClassDefFoundError: org/elasticsearch/action/admin/indices/analyze/AnalyzeRequest

### DISCUSSION THREAD

**Discussion by alawth** [Netflix/conductor#2808] [Q&A]
Hi,

I'm trying to connect 3.5.2 with aws elasticsearch for indexing. Following are the gradle dependancies within es-7 persistence;

implementation "org.elasticsearch:elasticsearch:7.10.0"
implementation "org.elasticsearch.client:elasticsearch-rest-client:7.10.0"
implementation "org.elasticsearch.client:elasticsearch-rest-high-level-client:7.10.0"

All configs are fine and es-7 module is getting loaded. On creating IndexDAO with ES7 following exception is occured.

Failed to instantiate [com.netflix.conductor.dao.IndexDAO]: Factory method 'es7IndexDAO' threw exception; nested exception is java.lang.NoClassDefFoundError: org/elasticsearch/action/admin/indices/analyze/AnalyzeRequest

Once I checked the class is was present in es-6.8.0 version and not available on latest versions. Any one got solved this issue?

Thanks.

**Comment by apanicker-nflx** [comment]
Please follow all the steps listed [here](https://github.com/Netflix/conductor/blob/main/es7-persistence/README.md) to enable using ES7.

**Reply by lzslim** [reply]
Hi, I follow the [doc ](https://github.com/Netflix/conductor/blob/main/es7-persistence/README.md)you posted, but after I changed the files as the doc suggest, recompiling the source code, it ended up with following error:

`> Configure project :
Inferred project: conductor, version: 3.6.0-SNAPSHOT

> Task :conductor-es6-persistence:compileJava FAILED
/home/lizheng/conductor/es6-persistence/src/main/java/com/netflix/conductor/es6/dao/index/ElasticSearchRestDAOV6.java:78: error: package com.fasterxml.jackson.databind does not exist
import com.fasterxml.jackson.databind.JsonNode;
                                     ^
/home/lizheng/conductor/es6-persistence/src/main/java/com/netflix/conductor/es6/dao/index/ElasticSearchRestDAOV6.java:79: error: package com.fasterxml.jackson.databind does not exist
import com.fasterxml.jackson.databind.ObjectMapper;
                                     ^
/home/lizheng/conductor/es6-persistence/src/main/java/com/netflix/conductor/es6/dao/index/ElasticSearchRestDAOV6.java:80: error: package com.fasterxml.jackson.databind.node does not exist
import com.fasterxml.jackson.databind.node.ObjectNode;
                                          ^
/home/lizheng/conductor/es6-persistence/src/main/java/com/netflix/conductor/es6/dao/index/ElasticSearchRestDAOV6.java:81: error: package com.fasterxml.jackson.databind.type does not exist
import com.fasterxml.jackson.databind.type.MapType;
                                          ^
/home/lizheng/conductor/es6-persistence/src/main/java/com/netflix/conductor/es6/dao/index/ElasticSearchRestDAOV6.java:82: error: package com.fasterxml.jackson.databind.type does not exist
import com.fasterxml.jackson.databind.type.TypeFactory;
                                          ^
/home/lizheng/conductor/es6-persistence/src/main/java/com/netflix/conductor/es6/dao/index/ElasticSearchRestDAOV6.java:123: error: cannot find symbol
    private final ObjectMapper objectMapper;
                  ^
  symbol:   class ObjectMapper
  location: class ElasticSearchRestDAOV6
/home/lizheng/conductor/es6-persistence/src/main/java/com/netflix/conductor/es6/dao/index/ElasticSearchRestDAOV6.java:140: error: cannot find symbol
            ObjectMapper objectMapper) {
            ^
  symbol:   class ObjectMapper
  location: class ElasticSearchRestDAOV6
/home/lizheng/conductor/es6-persistence/src/main/java/com/netflix/conductor/es6/dao/index/ElasticSearchDAOV6.java:74: error: package com.fasterxml.jackson.databind does not exist
import com.fasterxml.jackson.databind.ObjectMapper;
                                     ^
/home/lizheng/conductor/es6-persistence/src/main/java/com/netflix/conductor/es6/dao/index/ElasticSearchDAOV6.java:75: error: package com.fasterxml.jackson.databind.type does not exist
import com.fasterxml.jackson.databind.type.MapType;
                                          ^
/home/lizheng/conductor/es6-persistence/src/main/java/com/netflix/conductor/es6/dao/index/ElasticSearchDAOV6.java:76: error: package com.fasterxml.jackson.databind.type does not exist
import com.fasterxml.jackson.databind.type.TypeFactory;
                                          ^
/home/lizheng/conductor/es6-persistence/src/main/java/com/netflix/conductor/es6/dao/index/ElasticSearchDAOV6.java:106: error: cannot find symbol
    private final ObjectMapper objectMapper;
                  ^
  symbol:   class ObjectMapper
  location: class ElasticSearchDAOV6
/home/lizheng/conductor/es6-persistence/src/main/java/com/netflix/conductor/es6/dao/index/ElasticSearchDAOV6.java:127: error: cannot find symbol
            ObjectMapper objectMapper) {
            ^
  symbol:   class ObjectMapper
  location: class ElasticSearchDAOV6
/home/lizheng/conductor/es6-persistence/src/main/java/com/netflix/conductor/es6/config/ElasticSearchV6Configuration.java:42: error: package com.fasterxml.jackson.databind does not exist
import com.fasterxml.jackson.databind.ObjectMapper;
                                     ^
/home/lizheng/conductor/es6-persistence/src/main/java/com/netflix/conductor/es6/config/ElasticSearchV6Configuration.java:103: error: cannot find symbol
            ObjectMapper objectMapper) {
            ^
  symbol:   class ObjectMapper
  location: class ElasticSearchV6Configuration
/home/lizheng/conductor/es6-persistence/src/main/java/com/netflix/conductor/es6/config/ElasticSearchV6Configuration.java:110: error: cannot find symbol
            Client client, ElasticSearchProperties properties, ObjectMapper objectMapper) {
                                                               ^
  symbol:   class ObjectMapper
  location: class ElasticSearchV6Configuration
15 errors`

could you please provid some suggestions about this?

**Reply by apanicker-nflx** [reply]
Please make sure to update the dependency locks and change the elasticsearch.version to revElasticSearch7.
